lib/chef/provider/package.rb

```diff
@@ -86,11 +86,10 @@ def action_install
           end
         end
 
-        # XXX: mutating the new resource is generally bad
-        @new_resource.version(versions_for_new_resource)
-
         converge_by(install_description) do
-          install_package(package_names_for_targets, versions_for_targets)
+          with_subclass_array_wrapper(package_names_for_targets, versions_for_targets) do |name, version|
```

Comment:
i kind of like that, `multipackage_api_wrapper` maybe?  and then maybe `use_multipackage_api` for the class method?